src/services/db.ts

```diff
@@ -77,6 +77,41 @@ export const OVoteDisplayMode = {
 export type VoteDisplayMode =
   (typeof OVoteDisplayMode)[keyof typeof OVoteDisplayMode];
 
+const OSwipeActionBase = {
+  None: "none",
+  Upvote: "upvote",
+  Downvote: "downvote",
+  Reply: "reply",
+  Save: "save",
+} as const;
+
+export const OSwipeActionPost = {
+  ...OSwipeActionBase,
+  Hide: "hide",
+} as const;
+
+export const OSwipeActionComment = {
+  ...OSwipeActionBase,
+  Collapse: "collapse",
+} as const;
+
+export const OSwipeActionInbox = {
+  ...OSwipeActionBase,
+  MarkUnread: "mark_unread",
+} as const;
+
+export const OSwipeActionAll = {
+  ...OSwipeActionPost,
+  ...OSwipeActionComment,
+  ...OSwipeActionInbox,
+} as const;
+
+export type SwipeAction =
+  (typeof OSwipeActionAll)[keyof typeof OSwipeActionAll];
+
+export type SwipeDirection = "farStart" | "start" | "end" | "farEnd";
+export type SwipeActions = { [id in SwipeDirection]: SwipeAction };
```

Comment:
Can we change this to record notation? `Record<SwipeDirection, SwipeAction>`

There's probably an eslint rule for this somewhere I could add, too...